### PR TITLE
Add org.nextcloud.Nextcloud

### PR DESCRIPTION
**Pull request by juliushaertl**
Add build manifest for the Nextcloud sync client

### CODE REVIEW

File: org.nextcloud.Nextcloud.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            ],
+            "sources": [
+                {
+                    "type": "archive", 
+                    "url": "https://github.com/frankosterfeld/qtkeychain/archive/v0.8.0.tar.gz",
+                    "sha256": "b492f603197538bc04b2714105b1ab2b327a9a98d400d53d9a7cb70edd2db12f"
+                }            ]
+        },
+        {
+            "name": "nextcloud-client",
```
Comment (TingPing) [inline org.nextcloud.Nextcloud.json]: Needs an appdata file.

File: org.nextcloud.Nextcloud.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            "buildsystem": "cmake",
+            "config-opts": [
+                "-DCMAKE_BUILD_TYPE=Release",
+                "-DCMAKE_INSTALL_PREFIX=/app",
```
Comment (TingPing) [inline org.nextcloud.Nextcloud.json]: prefix should already be set.

File: org.nextcloud.Nextcloud.json [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    "command": "nextcloud",
+    "rename-desktop-file": "nextcloud.desktop",
+    "rename-icon": "Nextcloud",
+    "copy-icon": true,	
+    "finish-args": [
+        "--socket=x11",
+        "--share=network",
+        "--share=ipc",
+        "--filesystem=home:rw",
+        "--env=LD_LIBRARY_PATH=/app/lib:/app/lib/nextcloud",
```
Comment (TingPing) [inline org.nextcloud.Nextcloud.json]: `/app/lib` should already be in path

File: org.nextcloud.Nextcloud.json [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        "--talk-name=org.freedesktop.secrets"
+    ],
+    "modules": [
+        {
+            "name": "libsecret",
+            "config-opts": ["--disable-static", "--disable-gtk-doc", "--disable-manpages"],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "git://git.gnome.org/libsecret"
```
Comment (TingPing) [inline org.nextcloud.Nextcloud.json]: This needs to be a stable release.

File: org.nextcloud.Nextcloud.json [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+                "-DCMAKE_INSTALL_LIBDIR=/app/lib",
+                "-DLIB_INSTALL_DIR=/app/lib",
+                "-DBUILD_TRANSLATIONS=NO"
+            ],
+            "sources": [
+                {
+                    "type": "archive", 
+                    "url": "https://github.com/frankosterfeld/qtkeychain/archive/v0.8.0.tar.gz",
+                    "sha256": "b492f603197538bc04b2714105b1ab2b327a9a98d400d53d9a7cb70edd2db12f"
+                }            ]
```
Comment (TingPing) [inline org.nextcloud.Nextcloud.json]: Formatting nitpick

**Comment by juliushaertl** [comment]
@TingPing Thanks for the comments. I addressed them in the recent commit.

**Comment by TingPing** [comment]
Seems good, waiting on the KDE Sdk to land in flathub.

**Comment by nedrichards** [comment]
KDE SDK has now landed!

**Comment by Barthalion** [comment]
Started a test build: https://flathub.org/builds/#/builders/1/builds/381

**Comment by juliushaertl** [comment]
@Barthalion There was a naming issue with my appdata file. Should be fine now, from my understanding.

**Comment by TingPing** [comment]
Test build: https://flathub.org/builds/#/buildrequests/2200?redirect_to_build=true

File: org.nextcloud.Nextcloud.appdata.xml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Barthalion) [inline org.nextcloud.Nextcloud.appdata.xml]: You need to install this file manually. Look at RetroArch or Spotify (on Flathub) for examples.

**Comment by Barthalion** [comment]
https://flathub.org/builds/#/builders/1/builds/420

**Comment by TingPing** [comment]
Repository has been created: https://github.com/flathub/org.nextcloud.Nextcloud

**Comment by DanScharon** [comment]
I can't install the kde runtime, flatpak says:

`Required runtime for org.nextcloud.Nextcloud/x86_64/stable (org.kde.Platform/x86_64/5.9) is not installed, searching...
Found in remote flathub, do you want to install it? [y/n]: y
Installing: org.kde.Platform/x86_64/5.9 from flathub

1 metadata, 0 content objects fetched; 569 B transferred in 0 seconds                                                                                                                                                
error: While trying to checkout 8919553c6a8f3a6efb3b6363102185a0b48ef44559cd2a3632811be775b42875 into /home/<username>/.local/share/flatpak/runtime/org.kde.Platform/x86_64/5.9/.8919553c6a8f3a6efb3b6363102185a0b48ef44559cd2a3632811be775b42875-VIW36Y: Couldn't find file object '77235a0f2e8e54a81543d25e83b7174d7892e4487ea1020e853e2a62a7e3b29e'`

**Comment by TingPing** [comment]
Perhaps just a temporary problem? It works fine here with `flatpak` *0.9.12*

**Comment by DanScharon** [comment]
yes, it's working fine now. Thank you and sorry!

**Pull request closed without merging**